### ISSUE TITLE
jst.js being removed from layout.ejs on sails lift

### ISSUE THREAD

**Issue by JoaoCnh** [balderdashy/sails#3367]
I have defined in my pipeline.js to compile .ejs templates.

When I do that change it compiles and it adds it to layout.ejs

But if I do sails lift it removes it from layout.ejs therefore the JST templating stops working.

Anyone know why this happens?

**Comment by obj63mc** [comment]
Don't know if this is the correct way to fix this or not or if there is another way, but just had the same issue.  What I did was after 'compileAssets' in /tasks/register/build.js or /tasks/register/prod.js (if using sails lift --prod) I added in to run the 'syncAssets' task and this fixed the issue.  So my build.js file contains - 

        module.exports = function (grunt) {
        	grunt.registerTask('build', [
        		'compileAssets',
                'syncAssets',
        		'linkAssetsBuild',
        		'clean:build',
        		'copy:build'
        	]);
        };

This resolves the issue for me at least, would like to hear from the sails.js team if that is what really should be done before submitting a pull request.

**Comment by psdao1102** [comment]
I had the same issue. Heres the easier way to fix it. Go to the compileAssets task, and make sure that jst:dev is after copy:dev. It does not come this way by default, but this fixed the issue for me. I would like to see this change, (or something better) in a future release.

**Comment by justincy** [comment]
I hit this too. Spent much time on it. This ought to be bug.

**Comment by mikermcneil** [comment]
@JoaoCnh @justincy sorry to hear that- what version of Sails are you seeing this on?  We're currently developing using JST templates in Treeline using the latest Sails from master, but we're also using v0.11.x in prod and haven't run into this particular issue.

@obj63mc @psdao1102 thanks for posting your solutions!

**Comment by justincy** [comment]
I'm using v0.11.3

**Comment by justincy** [comment]
@psdao1102 I'm not convinced that is the best solution. I have to save my template files twice to refresh the compiled JST template.

**Comment by justincy** [comment]
The solution @obj63mc shared didn't work for me. I kept the change recommended by @psdao1102 and also changed `/tasks/register/syncAssets.js` to add `copy:dev` before `jst:dev` so that the files get copied before being compiled in the watch task (as happens in the initial build).

**Comment by mikermcneil** [comment]
Thanks for the follow-up @justincy.

Hmm- I haven't seen anything like this anywhere in any of the 0.11.x releases, so I'm assuming something may be going on here with either (a) versions of grunt dependencies locally installed in the actual sails project or (b) there's something I'm missing about the problem.

@ those of you experiencing this issue:

re: (a) -- would you mind pasting the dependencies section of your app's package.json here? Specifically, I'm looking for the grunt deps  (even better, if, after that, you could also run `npm ls` and paste the results here, that would be helpful to check and see if there are discrepancies between our versions)
re: (b) -- just to clarify, are we talking about the default setup?  We usually write client-side JST templates like `assets/templates/foo.html`.

Thanks!

**Comment by justincy** [comment]
Yes, I'm using the default setup.

package.json:

```js
  {
    "bcryptjs": "^2.3.0",
    "connect-redis": "1.4.5",
    "ejs": "~0.8.4",
    "grunt": "0.4.2",
    "grunt-contrib-clean": "~0.5.0",
    "grunt-contrib-coffee": "~0.10.1",
    "grunt-contrib-concat": "~0.3.0",
    "grunt-contrib-copy": "~0.5.0",
    "grunt-contrib-cssmin": "~0.9.0",
    "grunt-contrib-jst": "~0.6.0",
    "grunt-contrib-less": "0.11.1",
    "grunt-contrib-uglify": "~0.4.0",
    "grunt-contrib-watch": "~0.5.3",
    "grunt-sails-linker": "~0.9.5",
    "grunt-sync": "~0.0.4",
    "include-all": "~0.1.3",
    "nodemon": "^1.8.1",
    "passport": "^0.3.2",
    "passport-http-bearer": "^1.0.1",
    "passport-local": "^1.0.0",
    "rc": "~0.5.0",
    "sails": "~0.11.3",
    "sails-disk": "~0.10.0",
    "sails-generate-auth": "^0.3.1",
    "validator": "^4.5.0"
  }
```

npm ls:

```
├── bcrypt@0.8.5 extraneous
├── bcryptjs@2.3.0
├─┬ connect-redis@1.4.5
│ ├─┬ debug@2.2.0
│ │ └── ms@0.7.1
│ └── redis@0.7.3
├── ejs@0.8.8 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/ejs
├── forever@0.15.1 extraneous
├─┬ grunt@0.4.2 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt
│ ├── async@0.1.22
│ ├── coffee-script@1.3.3
│ ├── UNMET DEPENDENCY colors@~0.6.2
│ ├── dateformat@1.0.2-1.2.3
│ ├── eventemitter2@0.4.14
│ ├── exit@0.1.2
│ ├─┬ findup-sync@0.1.3
│ │ ├─┬ glob@3.2.11
│ │ │ ├── inherits@2.0.1
│ │ │ └─┬ minimatch@0.3.0
│ │ │   ├── lru-cache@2.7.3
│ │ │   └── sigmund@1.0.1
│ │ └── lodash@2.4.2
│ ├── getobject@0.1.0
│ ├─┬ glob@3.1.21
│ │ ├── graceful-fs@1.2.3
│ │ └── inherits@1.0.2
│ ├── hooker@0.2.3
│ ├── iconv-lite@0.2.11
│ ├─┬ js-yaml@2.0.5
│ │ ├─┬ argparse@0.1.16
│ │ │ ├── underscore@1.7.0
│ │ │ └── underscore.string@2.4.0
│ │ └── esprima@1.0.4
│ ├── lodash@0.9.2
│ ├─┬ minimatch@0.2.14
│ │ ├── lru-cache@2.7.3
│ │ └── sigmund@1.0.1
│ ├─┬ nopt@1.0.10
│ │ └── abbrev@1.0.7
│ ├─┬ rimraf@2.0.3
│ │ └── graceful-fs@1.1.14
│ ├── underscore.string@2.2.1
│ └── which@1.0.9
├─┬ grunt-contrib-clean@0.5.0 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-contrib-clean
│ ├── UNMET DEPENDENCY grunt@~0.4.0
│ └── rimraf@2.2.8
├─┬ grunt-contrib-coffee@0.10.1 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-contrib-coffee
│ ├─┬ chalk@0.4.0
│ │ ├── ansi-styles@1.0.0
│ │ ├── has-color@0.1.7
│ │ └── strip-ansi@0.1.1
│ ├─┬ coffee-script@1.7.1
│ │ └── mkdirp@0.3.5
│ ├── UNMET DEPENDENCY grunt@~0.4.0
│ └── UNMET DEPENDENCY lodash@~2.4.1
├─┬ grunt-contrib-concat@0.3.0 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-contrib-concat
│ └── UNMET DEPENDENCY grunt@~0.4.0
├─┬ grunt-contrib-copy@0.5.0 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-contrib-copy
│ └── UNMET DEPENDENCY grunt@~0.4.0
├─┬ grunt-contrib-cssmin@0.9.0 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-contrib-cssmin
│ ├─┬ chalk@0.4.0
│ │ ├── ansi-styles@1.0.0
│ │ ├── has-color@0.1.7
│ │ └── strip-ansi@0.1.1
│ ├─┬ clean-css@2.1.8
│ │ └── UNMET DEPENDENCY commander@2.1.x
│ ├── UNMET DEPENDENCY grunt@~0.4.1
│ └─┬ maxmin@0.1.0
│   ├─┬ gzip-size@0.1.1
│   │ ├─┬ concat-stream@1.5.1
│   │ │ ├── inherits@2.0.1
│   │ │ ├─┬ readable-stream@2.0.5
│   │ │ │ ├── core-util-is@1.0.2
│   │ │ │ ├── isarray@0.0.1
│   │ │ │ ├── process-nextick-args@1.0.6
│   │ │ │ ├── string_decoder@0.10.31
│   │ │ │ └── util-deprecate@1.0.2
│   │ │ └── typedarray@0.0.6
│   │ └─┬ zlib-browserify@0.0.3
│   │   └─┬ tape@0.2.2
│   │     ├── deep-equal@0.0.0
│   │     ├── defined@0.0.0
│   │     └── jsonify@0.0.0
│   └── pretty-bytes@0.1.2
├─┬ grunt-contrib-jst@0.6.0 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-contrib-jst
│ ├─┬ chalk@0.4.0
│ │ ├── ansi-styles@1.0.0
│ │ ├── has-color@0.1.7
│ │ └── strip-ansi@0.1.1
│ ├── UNMET DEPENDENCY grunt@~0.4.0
│ ├─┬ grunt-lib-contrib@0.7.1
│ │ ├─┬ maxmin@0.1.0
│ │ │ ├─┬ gzip-size@0.1.1
│ │ │ │ ├─┬ concat-stream@1.5.1
│ │ │ │ │ ├── inherits@2.0.1
│ │ │ │ │ ├─┬ readable-stream@2.0.5
│ │ │ │ │ │ ├── core-util-is@1.0.2
│ │ │ │ │ │ ├── isarray@0.0.1
│ │ │ │ │ │ ├── process-nextick-args@1.0.6
│ │ │ │ │ │ ├── string_decoder@0.10.31
│ │ │ │ │ │ └── util-deprecate@1.0.2
│ │ │ │ │ └── typedarray@0.0.6
│ │ │ │ └─┬ zlib-browserify@0.0.3
│ │ │ │   └─┬ tape@0.2.2
│ │ │ │     ├── deep-equal@0.0.0
│ │ │ │     ├── defined@0.0.0
│ │ │ │     └── jsonify@0.0.0
│ │ │ └── pretty-bytes@0.1.2
│ │ └── strip-path@0.1.1
│ └── UNMET DEPENDENCY lodash@~2.4.1
├─┬ grunt-contrib-less@0.11.1 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-contrib-less
│ ├── async@0.2.10
│ ├─┬ chalk@0.4.0
│ │ ├── ansi-styles@1.0.0
│ │ ├── has-color@0.1.7
│ │ └── strip-ansi@0.1.1
│ ├── UNMET DEPENDENCY grunt@^0.4.0
│ ├─┬ less@1.7.5
│ │ ├─┬ clean-css@2.2.23
│ │ │ └── commander@2.2.0
│ │ ├── graceful-fs@3.0.8
│ │ ├── mime@1.2.11
│ │ ├─┬ mkdirp@0.5.1
│ │ │ └── minimist@0.0.8
│ │ ├─┬ request@2.40.0
│ │ │ ├── aws-sign2@0.5.0
│ │ │ ├── forever-agent@0.5.2
│ │ │ ├─┬ form-data@0.1.4
│ │ │ │ ├── async@0.9.2
│ │ │ │ └─┬ combined-stream@0.0.7
│ │ │ │   └── delayed-stream@0.0.5
│ │ │ ├─┬ hawk@1.1.1
│ │ │ │ ├── boom@0.4.2
│ │ │ │ ├── cryptiles@0.2.2
│ │ │ │ ├── hoek@0.9.1
│ │ │ │ └── sntp@0.2.4
│ │ │ ├─┬ http-signature@0.10.1
│ │ │ │ ├── asn1@0.1.11
│ │ │ │ ├── assert-plus@0.1.5
│ │ │ │ └── ctype@0.5.3
│ │ │ ├── json-stringify-safe@5.0.1
│ │ │ ├── mime-types@1.0.2
│ │ │ ├── UNMET DEPENDENCY node-uuid@~1.4.0
│ │ │ ├── oauth-sign@0.3.0
│ │ │ ├── qs@1.0.2
│ │ │ ├── stringstream@0.0.5
│ │ │ ├── tough-cookie@2.2.1
│ │ │ └── tunnel-agent@0.4.2
│ │ └─┬ source-map@0.1.43
│ │   └── amdefine@1.0.0
│ ├── UNMET DEPENDENCY lodash@^2.4.1
│ └─┬ maxmin@0.1.0
│   ├─┬ gzip-size@0.1.1
│   │ ├─┬ concat-stream@1.5.1
│   │ │ ├── inherits@2.0.1
│   │ │ ├─┬ readable-stream@2.0.5
│   │ │ │ ├── core-util-is@1.0.2
│   │ │ │ ├── isarray@0.0.1
│   │ │ │ ├── process-nextick-args@1.0.6
│   │ │ │ ├── string_decoder@0.10.31
│   │ │ │ └── util-deprecate@1.0.2
│   │ │ └── typedarray@0.0.6
│   │ └─┬ zlib-browserify@0.0.3
│   │   └─┬ tape@0.2.2
│   │     ├── deep-equal@0.0.0
│   │     ├── defined@0.0.0
│   │     └── jsonify@0.0.0
│   └── pretty-bytes@0.1.2
├─┬ grunt-contrib-uglify@0.4.1 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-contrib-uglify
│ ├─┬ chalk@0.4.0
│ │ ├── ansi-styles@1.0.0
│ │ ├── has-color@0.1.7
│ │ └── strip-ansi@0.1.1
│ ├── UNMET DEPENDENCY grunt@~0.4.0
│ ├─┬ maxmin@0.1.0
│ │ ├─┬ gzip-size@0.1.1
│ │ │ ├─┬ concat-stream@1.5.1
│ │ │ │ ├── inherits@2.0.1
│ │ │ │ ├─┬ readable-stream@2.0.5
│ │ │ │ │ ├── core-util-is@1.0.2
│ │ │ │ │ ├── isarray@0.0.1
│ │ │ │ │ ├── process-nextick-args@1.0.6
│ │ │ │ │ ├── string_decoder@0.10.31
│ │ │ │ │ └── util-deprecate@1.0.2
│ │ │ │ └── typedarray@0.0.6
│ │ │ └─┬ zlib-browserify@0.0.3
│ │ │   └─┬ tape@0.2.2
│ │ │     ├── deep-equal@0.0.0
│ │ │     ├── defined@0.0.0
│ │ │     └── jsonify@0.0.0
│ │ └── pretty-bytes@0.1.2
│ └─┬ uglify-js@2.6.1
│   ├── async@0.2.10
│   ├── source-map@0.5.3
│   ├── uglify-to-browserify@1.0.2
│   └─┬ yargs@3.10.0
│     ├── camelcase@1.2.1
│     ├─┬ cliui@2.1.0
│     │ ├─┬ center-align@0.1.2
│     │ │ ├─┬ align-text@0.1.3
│     │ │ │ ├─┬ kind-of@2.0.1
│     │ │ │ │ └── is-buffer@1.1.1
│     │ │ │ ├── longest@1.0.1
│     │ │ │ └── repeat-string@1.5.2
│     │ │ └── lazy-cache@0.2.7
│     │ ├─┬ right-align@0.1.3
│     │ │ └─┬ align-text@0.1.3
│     │ │   ├─┬ kind-of@2.0.1
│     │ │   │ └── is-buffer@1.1.1
│     │ │   ├── longest@1.0.1
│     │ │   └── repeat-string@1.5.2
│     │ └── wordwrap@0.0.2
│     ├─┬ decamelize@1.1.2
│     │ └── escape-string-regexp@1.0.4
│     └── window-size@0.1.0
├─┬ grunt-contrib-watch@0.5.3 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-contrib-watch
│ ├─┬ gaze@0.4.3
│ │ └─┬ globule@0.1.0
│ │   ├─┬ glob@3.1.21
│ │   │ ├── graceful-fs@1.2.3
│ │   │ └── inherits@1.0.2
│ │   ├── lodash@1.0.2
│ │   └─┬ minimatch@0.2.14
│ │     ├── lru-cache@2.7.3
│ │     └── sigmund@1.0.1
│ ├── UNMET DEPENDENCY grunt@~0.4.0
│ └─┬ tiny-lr@0.0.4
│   ├── debug@0.7.4
│   ├── faye-websocket@0.4.4
│   ├─┬ noptify@0.0.3
│   │ └─┬ nopt@2.0.0
│   │   └── abbrev@1.0.7
│   └── qs@0.5.6
├── grunt-sails-linker@0.9.6 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-sails-linker
├─┬ grunt-sync@0.0.8 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/grunt-sync
│ └── promised-io@0.3.3
├─┬ include-all@0.1.6 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/include-all
│ └── underscore.string@2.3.1
├─┬ mocha@2.3.4
│ ├── commander@2.3.0
│ ├─┬ debug@2.2.0
│ │ └── ms@0.7.1
│ ├── diff@1.4.0
│ ├── escape-string-regexp@1.0.2
│ ├─┬ glob@3.2.3
│ │ ├── graceful-fs@2.0.3
│ │ ├── inherits@2.0.1
│ │ └─┬ minimatch@0.2.14
│ │   ├── lru-cache@2.7.3
│ │   └── sigmund@1.0.1
│ ├── growl@1.8.1
│ ├─┬ jade@0.26.3
│ │ ├── commander@0.6.1
│ │ └── mkdirp@0.3.0
│ ├─┬ mkdirp@0.5.0
│ │ └── minimist@0.0.8
│ └── supports-color@1.2.0
├─┬ nodemon@1.8.1
│ ├─┬ chokidar@1.4.2
│ │ ├─┬ anymatch@1.3.0
│ │ │ ├── arrify@1.0.1
│ │ │ └─┬ micromatch@2.3.7
│ │ │   ├─┬ arr-diff@2.0.0
│ │ │   │ └── arr-flatten@1.0.1
│ │ │   ├── array-unique@0.2.1
│ │ │   ├─┬ braces@1.8.3
│ │ │   │ ├─┬ expand-range@1.8.1
│ │ │   │ │ └─┬ fill-range@2.2.3
│ │ │   │ │   ├── is-number@2.1.0
│ │ │   │ │   ├─┬ isobject@2.0.0
│ │ │   │ │   │ └── isarray@0.0.1
│ │ │   │ │   ├── randomatic@1.1.5
│ │ │   │ │   └── repeat-string@1.5.2
│ │ │   │ ├── preserve@0.2.0
│ │ │   │ └── repeat-element@1.1.2
│ │ │   ├── expand-brackets@0.1.4
│ │ │   ├─┬ extglob@0.3.1
│ │ │   │ ├─┬ ansi-green@0.1.1
│ │ │   │ │ └── ansi-wrap@0.1.0
│ │ │   │ └── success-symbol@0.1.0
│ │ │   ├── filename-regex@2.0.0
│ │ │   ├── is-extglob@1.0.0
│ │ │   ├─┬ kind-of@3.0.2
│ │ │   │ └── is-buffer@1.1.1
│ │ │   ├── normalize-path@2.0.1
│ │ │   ├─┬ object.omit@2.0.0
│ │ │   │ ├─┬ for-own@0.1.3
│ │ │   │ │ └── for-in@0.1.4
│ │ │   │ └── is-extendable@0.1.1
│ │ │   ├─┬ parse-glob@3.0.4
│ │ │   │ ├── glob-base@0.3.0
│ │ │   │ └── is-dotfile@1.0.2
│ │ │   └─┬ regex-cache@0.4.2
│ │ │     ├── is-equal-shallow@0.1.3
│ │ │     └── is-primitive@2.0.0
│ │ ├── async-each@0.1.6
│ │ ├── glob-parent@2.0.0
│ │ ├── inherits@2.0.1
│ │ ├─┬ is-binary-path@1.0.1
│ │ │ └── binary-extensions@1.4.0
│ │ ├─┬ is-glob@2.0.1
│ │ │ └── is-extglob@1.0.0
│ │ ├── path-is-absolute@1.0.0
│ │ └─┬ readdirp@2.0.0
│ │   ├── graceful-fs@4.1.2
│ │   ├─┬ minimatch@2.0.10
│ │   │ └─┬ brace-expansion@1.1.2
│ │   │   ├── balanced-match@0.3.0
│ │   │   └── concat-map@0.0.1
│ │   └─┬ readable-stream@2.0.5
│ │     ├── core-util-is@1.0.2
│ │     ├── isarray@0.0.1
│ │     ├── process-nextick-args@1.0.6
│ │     ├── string_decoder@0.10.31
│ │     └── util-deprecate@1.0.2
│ ├─┬ debug@2.2.0
│ │ └── ms@0.7.1
│ ├── es6-promise@3.0.2
│ ├─┬ lodash.defaults@3.1.2
│ │ ├─┬ lodash.assign@3.2.0
│ │ │ ├─┬ lodash._baseassign@3.2.0
│ │ │ │ └── lodash._basecopy@3.0.1
│ │ │ ├─┬ lodash._createassigner@3.1.1
│ │ │ │ ├── lodash._bindcallback@3.0.1
│ │ │ │ └── lodash._isiterateecall@3.0.9
│ │ │ └─┬ lodash.keys@3.1.2
│ │ │   ├── lodash._getnative@3.9.1
│ │ │   ├── lodash.isarguments@3.0.4
│ │ │   └── lodash.isarray@3.0.4
│ │ └── lodash.restparam@3.6.1
│ ├─┬ minimatch@3.0.0
│ │ └─┬ brace-expansion@1.1.2
│ │   ├── balanced-match@0.3.0
│ │   └── concat-map@0.0.1
│ ├─┬ ps-tree@1.0.1
│ │ └─┬ event-stream@3.3.2
│ │   ├── duplexer@0.1.1
│ │   ├── from@0.1.3
│ │   ├── map-stream@0.1.0
│ │   ├── pause-stream@0.0.11
│ │   ├── split@0.3.3
│ │   ├── stream-combiner@0.0.4
│ │   └── through@2.3.8
│ ├─┬ touch@1.0.0
│ │ └─┬ nopt@1.0.10
│ │   └── abbrev@1.0.7
│ ├── undefsafe@0.0.3
│ └─┬ update-notifier@0.5.0
│   ├─┬ chalk@1.1.1
│   │ ├── ansi-styles@2.1.0
│   │ ├── escape-string-regexp@1.0.4
│   │ ├─┬ has-ansi@2.0.0
│   │ │ └── ansi-regex@2.0.0
│   │ ├─┬ strip-ansi@3.0.0
│   │ │ └── ansi-regex@2.0.0
│   │ └── supports-color@2.0.0
│   ├─┬ configstore@1.4.0
│   │ ├── graceful-fs@4.1.2
│   │ ├─┬ mkdirp@0.5.1
│   │ │ └── minimist@0.0.8
│   │ ├── object-assign@4.0.1
│   │ ├── os-tmpdir@1.0.1
│   │ ├─┬ osenv@0.1.3
│   │ │ └── os-homedir@1.0.1
│   │ ├── uuid@2.0.1
│   │ ├─┬ write-file-atomic@1.1.4
│   │ │ ├── imurmurhash@0.1.4
│   │ │ └── slide@1.1.6
│   │ └─┬ xdg-basedir@2.0.0
│   │   └── os-homedir@1.0.1
│   ├── is-npm@1.0.0
│   ├─┬ latest-version@1.0.1
│   │ └─┬ package-json@1.2.0
│   │   ├─┬ got@3.3.1
│   │   │ ├─┬ duplexify@3.4.2
│   │   │ │ ├─┬ end-of-stream@1.0.0
│   │   │ │ │ └─┬ once@1.3.3
│   │   │ │ │   └── wrappy@1.0.1
│   │   │ │ └─┬ readable-stream@2.0.5
│   │   │ │   ├── core-util-is@1.0.2
│   │   │ │   ├── inherits@2.0.1
│   │   │ │   ├── isarray@0.0.1
│   │   │ │   ├── process-nextick-args@1.0.6
│   │   │ │   ├── string_decoder@0.10.31
│   │   │ │   └── util-deprecate@1.0.2
│   │   │ ├── infinity-agent@2.0.3
│   │   │ ├── is-redirect@1.0.0
│   │   │ ├── is-stream@1.0.1
│   │   │ ├── lowercase-keys@1.0.0
│   │   │ ├─┬ nested-error-stacks@1.0.2
│   │   │ │ └── inherits@2.0.1
│   │   │ ├── object-assign@3.0.0
│   │   │ ├── prepend-http@1.0.3
│   │   │ ├─┬ read-all-stream@3.0.1
│   │   │ │ ├─┬ pinkie-promise@1.0.0
│   │   │ │ │ └── pinkie@1.0.0
│   │   │ │ └─┬ readable-stream@2.0.5
│   │   │ │   ├── core-util-is@1.0.2
│   │   │ │   ├── inherits@2.0.1
│   │   │ │   ├── isarray@0.0.1
│   │   │ │   ├── process-nextick-args@1.0.6
│   │   │ │   ├── string_decoder@0.10.31
│   │   │ │   └── util-deprecate@1.0.2
│   │   │ └── timed-out@2.0.0
│   │   └─┬ registry-url@3.0.3
│   │     └─┬ rc@1.1.6
│   │       ├── deep-extend@0.4.0
│   │       ├── ini@1.3.4
│   │       ├── minimist@1.2.0
│   │       └── strip-json-comments@1.0.4
│   ├─┬ repeating@1.1.3
│   │ └─┬ is-finite@1.0.1
│   │   └── number-is-nan@1.0.0
│   ├─┬ semver-diff@2.1.0
│   │ └── semver@5.1.0
│   └─┬ string-length@1.0.1
│     └─┬ strip-ansi@3.0.0
│       └── ansi-regex@2.0.0
├─┬ passport@0.3.2
│ ├── passport-strategy@1.0.0
│ └── pause@0.0.1
├── passport-http@0.3.0 extraneous
├─┬ passport-http-bearer@1.0.1
│ └── passport-strategy@1.0.0
├─┬ passport-local@1.0.0
│ └── passport-strategy@1.0.0
├─┬ rc@0.5.5 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/rc
│ ├── deep-extend@0.2.11
│ ├── ini@1.3.4
│ ├── minimist@0.0.10
│ └── strip-json-comments@0.1.3
├─┬ sails@0.11.3 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails
│ ├─┬ anchor@0.10.5
│ │ ├─┬ geojsonhint@1.1.0
│ │ │ ├── colors@0.6.2
│ │ │ ├─┬ concat-stream@1.4.10
│ │ │ │ ├── inherits@2.0.1
│ │ │ │ ├─┬ readable-stream@1.1.13
│ │ │ │ │ ├── core-util-is@1.0.1
│ │ │ │ │ ├── isarray@0.0.1
│ │ │ │ │ └── string_decoder@0.10.31
│ │ │ │ └── typedarray@0.0.6
│ │ │ ├─┬ jsonlint-lines@1.6.0
│ │ │ │ ├── JSV@4.0.2
│ │ │ │ └─┬ nomnom@1.8.1
│ │ │ │   ├─┬ chalk@0.4.0
│ │ │ │   │ ├── ansi-styles@1.0.0
│ │ │ │   │ ├── has-color@0.1.7
│ │ │ │   │ └── strip-ansi@0.1.1
│ │ │ │   └── underscore@1.6.0
│ │ │ ├── minimist@1.1.1
│ │ │ └─┬ optimist@0.6.1
│ │ │   ├── minimist@0.0.10
│ │ │   └── wordwrap@0.0.3
│ │ ├── lodash@3.9.3
│ │ └── validator@3.41.2
│ ├── async@0.9.2
│ ├─┬ captains-log@0.11.11
│ │ ├── lodash@2.4.1
│ │ └─┬ rc@0.3.5
│ │   ├── deep-extend@0.2.11
│ │   ├── ini@1.1.0
│ │   └── minimist@0.0.10
│ ├── colors@0.6.2
│ ├── commander@2.1.0
│ ├─┬ connect@2.25.0
│ │ ├── basic-auth-connect@1.0.0
│ │ ├─┬ body-parser@1.6.7
│ │ │ ├── iconv-lite@0.4.4
│ │ │ ├─┬ on-finished@2.1.0
│ │ │ │ └── ee-first@1.0.5
│ │ │ ├── qs@2.2.2
│ │ │ └── raw-body@1.3.0
│ │ ├── bytes@1.0.0
│ │ ├─┬ compression@1.0.11
│ │ │ ├─┬ accepts@1.0.7
│ │ │ │ ├── mime-types@1.0.2
│ │ │ │ └── negotiator@0.4.7
│ │ │ ├── compressible@1.1.1
│ │ │ ├── on-headers@1.0.1
│ │ │ └── vary@1.0.1
│ │ ├─┬ connect-timeout@1.2.2
│ │ │ ├── ms@0.6.2
│ │ │ └── on-headers@1.0.1
│ │ ├── cookie-parser@1.3.2
│ │ ├─┬ csurf@1.4.1
│ │ │ └─┬ csrf@2.0.7
│ │ │   ├── base64-url@1.2.1
│ │ │   ├── rndm@1.1.1
│ │ │   └── scmp@1.0.0
│ │ ├─┬ debug@1.0.4
│ │ │ └── ms@0.6.2
│ │ ├── depd@0.4.4
│ │ ├─┬ errorhandler@1.1.1
│ │ │ ├─┬ accepts@1.0.7
│ │ │ │ ├── mime-types@1.0.2
│ │ │ │ └── negotiator@0.4.7
│ │ │ └── escape-html@1.0.1
│ │ ├─┬ express-session@1.7.6
│ │ │ ├── buffer-crc32@0.2.3
│ │ │ ├── on-headers@1.0.1
│ │ │ ├── parseurl@1.3.0
│ │ │ ├─┬ uid-safe@1.0.1
│ │ │ │ ├── base64-url@1.2.1
│ │ │ │ └─┬ mz@1.3.0
│ │ │ │   ├── native-or-bluebird@1.2.0
│ │ │ │   ├── thenify@3.1.1
│ │ │ │   └── thenify-all@1.6.0
│ │ │ └── utils-merge@1.0.0
│ │ ├─┬ finalhandler@0.1.0
│ │ │ └── escape-html@1.0.1
│ │ ├── fresh@0.2.2
│ │ ├── media-typer@0.2.0
│ │ ├─┬ method-override@2.1.3
│ │ │ ├── methods@1.1.0
│ │ │ ├── parseurl@1.3.0
│ │ │ └── vary@1.0.1
│ │ ├─┬ morgan@1.2.3
│ │ │ ├── basic-auth@1.0.0
│ │ │ └─┬ on-finished@2.1.0
│ │ │   └── ee-first@1.0.5
│ │ ├─┬ multiparty@3.3.1
│ │ │ ├─┬ readable-stream@1.1.13
│ │ │ │ ├── core-util-is@1.0.2
│ │ │ │ ├── inherits@2.0.1
│ │ │ │ ├── isarray@0.0.1
│ │ │ │ └── string_decoder@0.10.31
│ │ │ └── stream-counter@0.2.0
│ │ ├── on-headers@0.0.0
│ │ ├── parseurl@1.2.0
│ │ ├── pause@0.0.1
│ │ ├── qs@1.0.2
│ │ ├── response-time@2.0.0
│ │ ├── serve-favicon@2.0.1
│ │ ├─┬ serve-index@1.1.6
│ │ │ ├─┬ accepts@1.0.7
│ │ │ │ ├── mime-types@1.0.2
│ │ │ │ └── negotiator@0.4.7
│ │ │ ├── batch@0.5.1
│ │ │ └── parseurl@1.3.0
│ │ ├─┬ serve-static@1.5.4
│ │ │ ├── escape-html@1.0.1
│ │ │ ├── parseurl@1.3.0
│ │ │ ├─┬ send@0.8.5
│ │ │ │ ├── destroy@1.0.3
│ │ │ │ ├── mime@1.2.11
│ │ │ │ ├── ms@0.6.2
│ │ │ │ ├─┬ on-finished@2.1.0
│ │ │ │ │ └── ee-first@1.0.5
│ │ │ │ └── range-parser@1.0.3
│ │ │ └── utils-merge@1.0.0
│ │ ├─┬ type-is@1.3.2
│ │ │ └── mime-types@1.0.2
│ │ └── vhost@2.0.0
│ ├── connect-flash@0.1.1
│ ├── cookie@0.1.2
│ ├── cookie-signature@1.0.4
│ ├── ejs-locals@1.0.2npm ERR! missing: colors@~0.6.2, required by grunt@0.4.2
npm ERR! missing: grunt@~0.4.0, required by grunt-contrib-clean@0.5.0
npm ERR! missing: lodash@~2.4.1, required by grunt-contrib-coffee@0.10.1
npm ERR! missing: grunt@~0.4.0, required by grunt-contrib-coffee@0.10.1
npm ERR! missing: grunt@~0.4.0, required by grunt-contrib-concat@0.3.0
npm ERR! missing: grunt@~0.4.0, required by grunt-contrib-copy@0.5.0
npm ERR! missing: grunt@~0.4.1, required by grunt-contrib-cssmin@0.9.0
npm ERR! missing: commander@2.1.x, required by clean-css@2.1.8
npm ERR! missing: lodash@~2.4.1, required by grunt-contrib-jst@0.6.0
npm ERR! missing: grunt@~0.4.0, required by grunt-contrib-jst@0.6.0
npm ERR! missing: lodash@^2.4.1, required by grunt-contrib-less@0.11.1
npm ERR! missing: grunt@^0.4.0, required by grunt-contrib-less@0.11.1
npm ERR! missing: node-uuid@~1.4.0, required by request@2.40.0
npm ERR! missing: grunt@~0.4.0, required by grunt-contrib-uglify@0.4.1
npm ERR! missing: grunt@~0.4.0, required by grunt-contrib-watch@0.5.3
npm ERR! missing: lodash@~2.4.1, required by sails-disk@0.10.8
npm ERR! missing: fs-extra@~0.8.1, required by sails-disk@0.10.8
npm ERR! missing: lodash@~2.4.1, required by waterline-criteria@0.11.2
npm ERR! missing: lodash@~2.4.1, required by waterline-cursor@0.0.6
npm ERR! extraneous: bcrypt@0.8.5 /home/ubuntu/workspace/toptal/expenses/node_modules/bcrypt
npm ERR! extraneous: passport-http@0.3.0 /home/ubuntu/workspace/toptal/expenses/node_modules/passport-http
npm ERR! extraneous: forever@0.15.1 /home/ubuntu/workspace/toptal/expenses/node_modules/forever

│ ├─┬ express@3.16.0
│ │ ├── basic-auth@1.0.0
│ │ ├── buffer-crc32@0.2.3
│ │ ├─┬ commander@1.3.2
│ │ │ └── keypress@0.1.0
│ │ ├─┬ debug@1.0.4
│ │ │ └── ms@0.6.2
│ │ ├── depd@0.4.4
│ │ ├── escape-html@1.0.1
│ │ ├── fresh@0.2.2
│ │ ├── media-typer@0.2.0
│ │ ├── merge-descriptors@0.0.2
│ │ ├── methods@1.1.0
│ │ ├─┬ mkdirp@0.5.0
│ │ │ └── minimist@0.0.8
│ │ ├── parseurl@1.2.0
│ │ ├─┬ proxy-addr@1.0.1
│ │ │ └── ipaddr.js@0.1.2
│ │ ├── range-parser@1.0.0
│ │ ├─┬ send@0.8.1
│ │ │ ├─┬ finished@1.2.2
│ │ │ │ └── ee-first@1.0.3
│ │ │ ├── mime@1.2.11
│ │ │ └── ms@0.6.2
│ │ └── vary@0.1.0
│ ├─┬ express-handlebars@1.0.3
│ │ ├─┬ glob@4.5.3
│ │ │ ├─┬ inflight@1.0.4
│ │ │ │ └── wrappy@1.0.1
│ │ │ ├── inherits@2.0.1
│ │ │ ├─┬ minimatch@2.0.10
│ │ │ │ └─┬ brace-expansion@1.1.2
│ │ │ │   ├── balanced-match@0.3.0
│ │ │ │   └── concat-map@0.0.1
│ │ │ └─┬ once@1.3.3
│ │ │   └── wrappy@1.0.1
│ │ ├── graceful-fs@3.0.8
│ │ ├─┬ handlebars@2.0.0
│ │ │ ├─┬ optimist@0.3.7
│ │ │ │ └── wordwrap@0.0.3
│ │ │ └─┬ uglify-js@2.3.6
│ │ │   ├── async@0.2.10
│ │ │   └─┬ source-map@0.1.43
│ │ │     └── amdefine@1.0.0
│ │ ├─┬ promise@5.0.0
│ │ │ └── asap@1.0.0
│ │ └── semver@3.0.1
│ ├─┬ fs-extra@0.8.1
│ │ ├── jsonfile@1.1.1
│ │ ├── mkdirp@0.3.5
│ │ ├── ncp@0.4.2
│ │ └── rimraf@2.2.8
│ ├─┬ glob@3.2.11
│ │ ├── inherits@2.0.1
│ │ └─┬ minimatch@0.3.0
│ │   ├── lru-cache@2.7.3
│ │   └── sigmund@1.0.1
│ ├─┬ grunt-cli@0.1.13
│ │ ├── findup-sync@0.1.3
│ │ ├─┬ nopt@1.0.10
│ │ │ └── abbrev@1.0.7
│ │ └── resolve@0.3.1
│ ├─┬ i18n@0.5.0
│ │ ├─┬ debug@2.2.0
│ │ │ └── ms@0.7.1
│ │ ├── mustache@2.2.1
│ │ └── sprintf@0.1.5
│ ├── lodash@2.4.2
│ ├── merge-defaults@0.1.4
│ ├─┬ method-override@2.3.5
│ │ ├─┬ debug@2.2.0
│ │ │ └── ms@0.7.1
│ │ ├── methods@1.1.1
│ │ ├── parseurl@1.3.0
│ │ └── vary@1.0.1
│ ├── mock-req@0.1.0
│ ├── mock-res@0.1.0
│ ├── node-uuid@1.4.7
│ ├── pluralize@0.0.12
│ ├─┬ prompt@0.2.14
│ │ ├── pkginfo@0.3.1
│ │ ├─┬ read@1.0.7
│ │ │ └── mute-stream@0.0.5
│ │ ├── revalidator@0.1.8
│ │ ├─┬ utile@0.2.1
│ │ │ ├── async@0.2.10
│ │ │ ├── deep-equal@1.0.1
│ │ │ ├── i@0.3.3
│ │ │ ├─┬ mkdirp@0.5.1
│ │ │ │ └── minimist@0.0.8
│ │ │ ├── ncp@0.4.2
│ │ │ └─┬ rimraf@2.5.0
│ │ │   └─┬ glob@6.0.3
│ │ │     ├─┬ inflight@1.0.4
│ │ │     │ └── wrappy@1.0.1
│ │ │     ├── inherits@2.0.1
│ │ │     ├─┬ minimatch@3.0.0
│ │ │     │ └─┬ brace-expansion@1.1.2
│ │ │     │   ├── balanced-match@0.3.0
│ │ │     │   └── concat-map@0.0.1
│ │ │     ├─┬ once@1.3.3
│ │ │     │ └── wrappy@1.0.1
│ │ │     └── path-is-absolute@1.0.0
│ │ └─┬ winston@0.8.3
│ │   ├── async@0.2.10
│ │   ├── cycle@1.0.3
│ │   ├── eyes@0.1.8
│ │   ├── isstream@0.1.2
│ │   └── stack-trace@0.0.9
│ ├─┬ reportback@0.1.9
│ │ └── switchback@1.1.3
│ ├── sails-build-dictionary@0.10.1
│ ├─┬ sails-generate@0.12.4
│ │ ├── async@0.2.10
│ │ ├── sails-generate-adapter@0.10.5
│ │ ├── sails-generate-api@0.10.0
│ │ ├── sails-generate-backend@0.12.2
│ │ ├─┬ sails-generate-controller@0.10.8
│ │ │ ├── pluralize@0.0.9
│ │ │ └── underscore.string@2.3.3
│ │ ├─┬ sails-generate-frontend@0.11.6
│ │ │ └── sails-generate-sails.io.js@0.11.7
│ │ ├── sails-generate-generator@0.10.11
│ │ ├── sails-generate-gruntfile@0.10.10
│ │ ├─┬ sails-generate-model@0.10.10
│ │ │ └── underscore.string@2.3.3
│ │ ├─┬ sails-generate-new@0.10.22
│ │ │ ├── async@1.5.1
│ │ │ ├── lodash@3.10.1
│ │ │ ├─┬ merge-defaults@0.2.1
│ │ │ │ └── lodash@2.4.2
│ │ │ └── win-spawn@2.0.0
│ │ ├── sails-generate-sails.io.js@0.12.0
│ │ ├── sails-generate-views@0.10.5
│ │ └── sails-generate-views-jade@0.10.3
│ ├─┬ sails-hook-sockets@0.11.28
│ │ ├─┬ machinepack-urls@3.1.1
│ │ │ └─┬ machine@4.1.1
│ │ │   ├── convert-to-ecmascript-compatible-varname@0.1.5
│ │ │   ├─┬ debug@2.2.0
│ │ │   │ └── ms@0.7.1
│ │ │   ├── object-hash@0.3.0
│ │ │   ├── rttc@1.0.2
│ │ │   └── switchback@1.1.3
│ │ ├── semver@4.3.6
│ │ ├─┬ socket.io@1.3.7
│ │ │ ├─┬ debug@2.1.0
│ │ │ │ └── ms@0.6.2
│ │ │ ├─┬ engine.io@1.5.4
│ │ │ │ ├── base64id@0.1.0
│ │ │ │ ├─┬ debug@1.0.3
│ │ │ │ │ └── ms@0.6.2
│ │ │ │ ├─┬ engine.io-parser@1.2.2
│ │ │ │ │ ├── after@0.8.1
│ │ │ │ │ ├── arraybuffer.slice@0.0.6
│ │ │ │ │ ├── base64-arraybuffer@0.1.2
│ │ │ │ │ ├── blob@0.0.4
│ │ │ │ │ ├─┬ has-binary@0.1.6
│ │ │ │ │ │ └── isarray@0.0.1
│ │ │ │ │ └── utf8@2.1.0
│ │ │ │ └─┬ ws@0.8.0
│ │ │ │   ├─┬ bufferutil@1.2.1
│ │ │ │   │ ├── bindings@1.2.1
│ │ │ │   │ └── nan@2.1.0
│ │ │ │   ├── options@0.0.6
│ │ │ │   ├── ultron@1.0.2
│ │ │ │   └─┬ utf-8-validate@1.2.1
│ │ │ │     ├── bindings@1.2.1
│ │ │ │     └── nan@2.1.0
│ │ │ ├─┬ has-binary-data@0.1.3
│ │ │ │ └── isarray@0.0.1
│ │ │ ├─┬ socket.io-adapter@0.3.1
│ │ │ │ ├─┬ debug@1.0.2
│ │ │ │ │ └── ms@0.6.2
│ │ │ │ ├── object-keys@1.0.1
│ │ │ │ └─┬ socket.io-parser@2.2.2
│ │ │ │   ├── benchmark@1.0.0
│ │ │ │   ├── component-emitter@1.1.2
│ │ │ │   ├── debug@0.7.4
│ │ │ │   ├── isarray@0.0.1
│ │ │ │   └── json3@3.2.6
│ │ │ └─┬ socket.io-parser@2.2.4
│ │ │   ├── benchmark@1.0.0
│ │ │   ├── component-emitter@1.1.2
│ │ │   ├── debug@0.7.4
│ │ │   ├── isarray@0.0.1
│ │ │   └── json3@3.2.6
│ │ └─┬ socket.io-client@1.3.7
│ │   ├── backo2@1.0.2
│ │   ├── component-bind@1.0.0
│ │   ├── component-emitter@1.1.2
│ │   ├── debug@0.7.4
│ │   ├─┬ engine.io-client@1.5.4
│ │   │ ├── component-inherit@0.0.3
│ │   │ ├─┬ debug@1.0.4
│ │   │ │ └── ms@0.6.2
│ │   │ ├─┬ engine.io-parser@1.2.2
│ │   │ │ ├── after@0.8.1
│ │   │ │ ├── arraybuffer.slice@0.0.6
│ │   │ │ ├── base64-arraybuffer@0.1.2
│ │   │ │ ├── blob@0.0.4
│ │   │ │ └── utf8@2.1.0
│ │   │ ├─┬ has-cors@1.0.3
│ │   │ │ └── global@2.0.1
│ │   │ ├─┬ parsejson@0.0.1
│ │   │ │ └─┬ better-assert@1.0.2
│ │   │ │   └── callsite@1.0.0
│ │   │ ├─┬ parseqs@0.0.2
│ │   │ │ └─┬ better-assert@1.0.2
│ │   │ │   └── callsite@1.0.0
│ │   │ ├─┬ parseuri@0.0.4
│ │   │ │ └─┬ better-assert@1.0.2
│ │   │ │   └── callsite@1.0.0
│ │   │ ├─┬ ws@0.8.0
│ │   │ │ ├─┬ bufferutil@1.2.1
│ │   │ │ │ ├── bindings@1.2.1
│ │   │ │ │ └── nan@2.1.0
│ │   │ │ ├── options@0.0.6
│ │   │ │ ├── ultron@1.0.2
│ │   │ │ └─┬ utf-8-validate@1.2.1
│ │   │ │   ├── bindings@1.2.1
│ │   │ │   └── nan@2.1.0
│ │   │ └── xmlhttprequest@1.5.0
│ │   ├─┬ has-binary@0.1.6
│ │   │ └── isarray@0.0.1
│ │   ├── indexof@0.0.1
│ │   ├── object-component@0.0.3
│ │   ├─┬ parseuri@0.0.2
│ │   │ └─┬ better-assert@1.0.2
│ │   │   └── callsite@1.0.0
│ │   ├─┬ socket.io-parser@2.2.4
│ │   │ ├── benchmark@1.0.0
│ │   │ ├── isarray@0.0.1
│ │   │ └── json3@3.2.6
│ │   └── to-array@0.1.3
│ ├── sails-stringfile@0.3.2
│ ├─┬ sails-util@0.10.6
│ │ ├── json-stringify-safe@5.0.1
│ │ ├─┬ optimist@0.6.1
│ │ │ ├── minimist@0.0.10
│ │ │ └── wordwrap@0.0.3
│ │ ├── switchback@1.1.3
│ │ └── underscore.string@2.3.3
│ ├── semver@2.2.1
│ ├─┬ skipper@0.5.8
│ │ ├── async@1.4.2
│ │ ├── colors@1.1.2
│ │ ├─┬ debug@2.2.0
│ │ │ └── ms@0.7.1
│ │ ├── dot-access@1.0.0
│ │ ├─┬ multiparty@3.2.10
│ │ │ ├─┬ readable-stream@1.1.13
│ │ │ │ ├── core-util-is@1.0.2
│ │ │ │ ├── inherits@2.0.1
│ │ │ │ └── isarray@0.0.1
│ │ │ └── stream-counter@0.2.0
│ │ ├── semver@5.0.3
│ │ ├── skipper-disk@0.5.4
│ │ └── string_decoder@0.10.31
│ ├─┬ uid-safe@1.1.0
│ │ ├── base64-url@1.2.1
│ │ └── native-or-bluebird@1.1.2
│ └─┬ waterline@0.10.30
│   ├─┬ anchor@0.11.0
│   │ ├─┬ geojsonhint@1.1.0
│   │ │ ├── colors@0.6.2
│   │ │ ├─┬ concat-stream@1.4.10
│   │ │ │ ├── inherits@2.0.1
│   │ │ │ ├─┬ readable-stream@1.1.13
│   │ │ │ │ ├── core-util-is@1.0.2
│   │ │ │ │ ├── isarray@0.0.1
│   │ │ │ │ └── string_decoder@0.10.31
│   │ │ │ └── typedarray@0.0.6
│   │ │ ├─┬ jsonlint-lines@1.6.0
│   │ │ │ ├── JSV@4.0.2
│   │ │ │ └─┬ nomnom@1.8.1
│   │ │ │   ├─┬ chalk@0.4.0
│   │ │ │   │ ├── ansi-styles@1.0.0
│   │ │ │   │ ├── has-color@0.1.7
│   │ │ │   │ └── strip-ansi@0.1.1
│   │ │ │   └── underscore@1.6.0
│   │ │ ├── minimist@1.1.1
│   │ │ └─┬ optimist@0.6.1
│   │ │   ├── minimist@0.0.10
│   │ │   └── wordwrap@0.0.3
│   │ ├── lodash@3.10.1
│   │ └── validator@4.4.0
│   ├── async@1.2.1
│   ├── bluebird@2.9.34
│   ├── deep-diff@0.3.3
│   ├── lodash@3.9.3
│   ├─┬ switchback@2.0.0
│   │ └── lodash@2.4.2
│   ├─┬ waterline-criteria@0.11.2
│   │ └── lodash@2.4.2
│   └─┬ waterline-schema@0.1.19
│     └── lodash@3.10.1
├─┬ sails-disk@0.10.8 -> /home/ubuntu/.nvm/versions/node/v4.2.1/lib/node_modules/sails/node_modules/sails-disk
│ ├── async@0.2.10
│ ├── UNMET DEPENDENCY fs-extra@~0.8.1
│ ├── UNMET DEPENDENCY lodash@~2.4.1
│ ├─┬ waterline-criteria@0.11.2
│ │ └── UNMET DEPENDENCY lodash@~2.4.1
│ ├─┬ waterline-cursor@0.0.6
│ │ ├── async@0.9.2
│ │ └── UNMET DEPENDENCY lodash@~2.4.1
│ └── waterline-errors@0.10.1
├─┬ sails-generate-auth@0.3.1
│ ├── lodash@3.10.1
│ └─┬ merge-defaults@0.2.1
│   └── lodash@2.4.2
├─┬ sails-memory@0.10.5
│ ├── async@0.2.10
│ ├── lodash@2.4.2
│ ├── waterline-criteria@0.10.7
│ ├─┬ waterline-cursor@0.0.6
│ │ └── async@0.9.2
│ └── waterline-errors@0.10.1
├─┬ supertest@1.1.0
│ ├── methods@1.1.1
│ └─┬ superagent@1.3.0
│   ├── component-emitter@1.1.2
│   ├── cookiejar@2.0.1
│   ├─┬ debug@2.2.0
│   │ └── ms@0.7.1
│   ├── extend@1.2.1
│   ├─┬ form-data@0.2.0
│   │ ├── async@0.9.2
│   │ ├─┬ combined-stream@0.0.7
│   │ │ └── delayed-stream@0.0.5
│   │ └─┬ mime-types@2.0.14
│   │   └── mime-db@1.12.0
│   ├── formidable@1.0.14
│   ├── methods@1.0.1
│   ├── mime@1.3.4
│   ├── qs@2.3.3
│   ├─┬ readable-stream@1.0.27-1
│   │ ├── core-util-is@1.0.2
│   │ ├── inherits@2.0.1
│   │ ├── isarray@0.0.1
│   │ └── string_decoder@0.10.31
│   └── reduce-component@1.0.1
└── validator@4.5.0
```

**Comment by mikermcneil** [comment]
Thanks @justincy. re: @psdao1102's suggestion of putting `copy` before `jst`, I did a little looking into what's up here.  We're currently using the default asset pipeline and haven't noticed any issues with JST.  Our `compileAssets` task looks like so:

```javascript
module.exports = function(grunt) {
  grunt.registerTask('compileAssets', [
    'clean:dev',
    'jst:dev',
    'less:dev',
    'copy:dev',
    'coffee:dev'
  ]);
};
```

I'll follow up momentarily with the grunt dep versions we're using so if they're different, you can try those out make sure there aren't any issues.

In the mean time, a few more questions to help to get to the bottom of this:
- are you naming your template files with a `.html` suffix?  (eg. `assets/templates/foo/bar/baz.html`)
- is this only happening when Sails is lifted in production mode?  Does it happen when running `sails www`?  Does it happen when Sails is lifted in development?

**Comment by mikermcneil** [comment]
> I would like to see this change, (or something better) in a future release.

@psdao1102 would you provide some more info about the advantages to the approach?  I'm happy to make that happen asap as the default convention in the upcoming 0.12 release if you can help me understand more of the details and ensure there aren't associated downsides.  Luckily, changing this would be pretty easy -- i.e. since it's all in generators, it doesn't add a migration burden for existing Sails projects.

Either way, I'd like to understand your reasoning.  I respect that it can take some substantial effort to write up that analysis, so know that I don't ask you to do that lightly, and I appreciate your help. If you have some time to do that, it'd be awesome if you could open it as a short and sweet [feature proposal](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md#v-proposing-features-and-enhancements) and link to it from here.  No pressure- but I'll keep an eye out for it in the next few days in case the timing works. Thanks :+1:

**Comment by justincy** [comment]
> * are you naming your template files with a .html suffix? (eg. assets/templates/foo/bar/baz.html)

Yes.

> * is this only happening when Sails is lifted in production mode? Does it happen when running sails www? Does it happen when Sails is lifted in development?

I never tested production mode. This was happening to me in development mode.

**Comment by psdao1102** [comment]
Thanks for responding mikermcneil. I cannot say for sure but it seems to me that the jst task checks the public files for templates and not the ones in the assets folder. But to be fair the only reason why I have made this assumption was that my suggestion from fresh installed worked. If i had a little more time i would verify my findings

As for the other comment It happens in both production and development. And both when i use html or ejs (with the appropriate configuration changes).

**Comment by mikermcneil** [comment]
You know, something just occurred to me...

@psdao1102 @justincy @obj63mc @JoaoCnh What IDEs / text editors are you guys using?

**Comment by justincy** [comment]
Cloud9

**Comment by mikermcneil** [comment]
@psdao1102 @obj63mc @JoaoCnh What about you guys?

**Comment by mikermcneil** [comment]
(@justincy would you mind checking to see if this happens using emacs, vim, or Sublime Text?)

**Comment by mikermcneil** [comment]
Without more information, since we haven't been able to reproduce this, I'm going to go ahead and close.   I have a hunch this might have been solved by https://github.com/balderdashy/sails/issues/3545 -- or otherwise have something to do with text editors-- but I'd need more information to be able to help. If anyone continues to see the problem using 0.11.5 or 0.12.x, please open a new issue and let us know.  Thanks!
